src/Security/test/Authorization.Certificate.Test/CertificateHttpClientBuilderExtensionsTest.cs

```diff
@@ -31,11 +32,47 @@ public async Task AddCertificateAuthorizationClient_AddsNamedHttpClientWithCerti
         certificateHeader.Should().BeEquivalentTo(Convert.ToBase64String(certificateBytes));
     }
 
-    private static IHostBuilder GetHostBuilder()
+    [Fact]
+    public async Task AddCertificateAuthorizationClient_AllowsCustomHeader()
+    {
+        const string customCertificateHeader = "my-arbitrary-header";
+        byte[] certificateBytes = X509Certificate2.CreateFromPemFile("instance.crt", "instance.key").Export(X509ContentType.Cert);
+        using var appScope = new EnvironmentVariableScope("VCAP_APPLICATION", "not empty");
+        using var certScope = new EnvironmentVariableScope("CF_INSTANCE_CERT", "instance.crt");
+        using var keyScope = new EnvironmentVariableScope("CF_INSTANCE_KEY", "instance.key");
+        IHostBuilder builder = GetHostBuilder(customCertificateHeader);
+
+        builder.ConfigureServices(services => services.PostConfigure<CertificateForwardingOptions>(option =>
+        {
+            option.CertificateHeader = customCertificateHeader;
+        }));
```

Comment:
I'm a bit surprised to see this is needed. Can't Steeltoe take care of that?

Reply:
Not only does Steeltoe already do what is needed, `CertificateForwardingOptions` isn't relevant to this test at all because no requests are even sent... Good catch, this was a leftover from my exploration about configuring the header on the server-side and should be removed